spec/GraphQLOverHTTP.md

```diff
@@ -738,10 +734,15 @@ Note: The GraphQL specification
 and refers to the situation wherein a _GraphQL field error_ occurs as a partial
 response; it still indicates successful execution.
 
-## Processing the response
+## Processing a response
+
+The source of a response may be an intermediary server, such as an API gateway,
+proxy, or firewall, in certain environments.
+
+Those intermediary servers may not understand GraphQL: in the case of an error,
+they may return their own non-GraphQL `application/json` response with a
+non-`200` status code.
 
-If the response uses a non-`200` status code and the media type of the response
-payload is `application/json` then the client MUST NOT rely on the body to be a
-well-formed _GraphQL response_ since the source of the response may not be the
-server but instead some intermediary such as API gateways, proxies, firewalls,
-etc.
+For this reason, if the response uses a non-`200` status code and the media type
+of the response body is `application/json` then the client MUST NOT rely on the
+body to be a well-formed _GraphQL response_.
```

Comment:
I just want to point out that sentence defines a rule that the client must follow.  The specification is a server specification, correct?  Do we have any other client requirements within this spec?  Perhaps section this should be a non-normative note as to client behavior.

Reply:
Agreed, the client sentences feel a bit out of place. I'll update that.

Reply:
Updated [here](https://github.com/graphql/graphql-over-http/pull/304/commits/ddd4ba9ab9721b53b02faf49a08a8ac5f554e829). I also took this opportunity to remove the spec-like language (`MUST NOT rely on ...`) and replace it with pseudo-logic that I would implement in a client application. Let me know what you think.

Reply:
Looks good to me.  Normally I might write 'can only rely on' or something (with the word 'only') but that would preclude optional additional behaviors on the part of the server.

Reply:
I contemplated that 'only' for a while too and decided to leave it out ultimately for the same reason. What about adding a trailing sentence:

```
The other cases require the client to have knowledge about the specific service behavior 
and are outside the scope of this specification.
```
?

Reply:
Eh.  Whatever you think.  I'd probably skip it.

Reply:
Sounds good, let's keep it simple 👍 . Thanks for all the feedbacks 🙏